internal/core/metadata/rest_addressable_test.go

```diff
@@ -52,13 +52,9 @@ var ErrorPathParam = "%zz"
 // ErrorPortPathParam path parameter used to trigger an error in the `restGetAddressableByPort` function where the port variable is expected to be a number.
 var ErrorPortPathParam = "abc"
 
-func TestMain(m *testing.M) {
-	LoggingClient = logger.NewMockClient()
-	httpErrorHandler = errorconcept.NewErrorHandler(LoggingClient)
-	os.Exit(m.Run())
-}
-
 func TestGetAllAddressables(t *testing.T) {
+
+	httpErrorHandler = errorconcept.NewErrorHandler(logger.NewMockClient())
```

Comment:
Nitpick: you create a logger.NewMockClient() here and elsewhere in this test function.  You should only instantiate and use one through the test function.